### DISCUSSION TITLE
Cashio Revival Project Discussion & Feedback (Ouroboros DAO)

### DISCUSSION THREAD

**Discussion by ghost** [saber-hq/governance#1] [General]
We are now months past the Cashio Exploit. It is time for our community to assess and provide feedback on our plans to make everyone whole. **We would also like to put a special emphasis on the potential to flip the current Saber Wars incentive structure on its head.** 

> Our Revival Project will provide locked Protocol Owned Liquidity (POL) for Saber LPs, and also Protocol Owned Governance (POG) for SBR/SUNNY gauges. At the same time, it could make use of a Protocol Owned Market (POM) which redirects liquidity to a guaranteed floor price - which finally allows us to issue call option rewards to victims in a **sustainable manner**. Saber is also granting us their [Liquidity Reserve](https://blog.saber.so/introducing-sbr-the-saber-protocol-token-5f1f1103d57d) (842m SBR, 8.42% of SBR) vested over 1-2 years. This SBR is going to be permalocked in the protocol **(for a total of 8.42b veSBR at end of vest)** and will bootstrap the POG which can only be accessed by purchasing & locking the Treasury Token- thus contributing to a rising floor.

As for the CASH Exploit, we’ve been able to convince [the exploiter](https://etherscan.io/address/0x86766247ba3405c5f15f06b895294200809e9cfb) to return some funds (approx. $15m out of $40m claimed by users for refund via [our verification tool](https://anchor.so/cashio)) there is still $24.5m in outstanding refunds and $25m in the exploiter’s wallet. **Unfortunately, since we made our offer to the exploiter to keep the remaining ETH as a white hat bounty - the price of ETH has plummeted leaving them with <20% of the bounty they could have claimed just a few weeks earlier.**

Because of this fact it is unclear when or whether the funds will be returned - we have been focusing intensely on what we can do with the resources we could gather to provide a system with the highest probability of succeeding in making victims whole in a sustainable manner while the exploiters decide to do the right thing or are brought to justice by the authorities on the case.

**At the same time, we’ve entered an extreme bear market with L1s, large stable coins, and mostly everything else crashing.** Although this is unfortunate for most of us in the space, it highlights the need for diversifying sustainable, low risks investments & stores of value.

Furthermore, within the Saber Wars **there is a clear problem of mercenary farming where the $SBR token is currently only serving the purpose of being a reward/dumping token, since retail rarely votes (or cares about votes on gauges) compared to protocols.** 

Our solution: **Ouroboros**. A two token protocol that has OURO and BOROS, where OURO is a stablecoin minted from LPs - and BOROS is a Treasury Token that can be purchased from a POM with Ouro. veBoros is used to control the SBR/SUNNY in the Treasury, and to earn Call Option Rewards as sustainable APY for participating in the system. The Call Option Rewards allow users to purchase Boros at its floor price, and either sell it at the spot price or lock it for a higher veBOROS balance.
**TLDR, heres a diagram explaining the proposed system for the new protocol around launch:**        
![image](https://user-images.githubusercontent.com/101472787/170216470-5f434164-3c37-4f60-9990-cdc29e2ec405.png)

Below will be the short summary, and [if you are confused by unfamiliar terminology please check out this Key Terms page. ](https://cashio2.notion.site/Key-Terms-ff5c0467dd57457ea4b696e0f604e212)

The [notion site](https://cashio2.notion.site/cashio2/Cashio-Revival-Proposal-1c7f53c3a4f44f96b91ce2a52f0c5333) has more details, and will include the latest information from the development team as we finalize the exact functions people want to see in this protocol.

[Here](https://www.notion.so/cashio2/Open-Questions-For-Feedback-95c63948f6364b32a7a19db596258375), you can see the main open questions from the team regarding what we would like some feedback on.
     
**So with all that being said, here’s the things we hope to achieve with this revival:**

1. Create **sustainable** rewards for victims with unbounded upside but limited downside - through the use of floor value reserves & call option rewards.
2. **Provide locked, deep liquidity to other stable coins** through our neutral base pair & lock the liquidity using the veToken mechanics on our governance(”Treasury”) token (which acts as a sink for the base pair & distributor of value from the system using a floor price mechanic).
> In order to get rewarded or access governance using our token, you need to stake for a minimum of 7 days or maximum of 5 years. Staking for 7 days gives you a 1x vote for every token and 1x rewards, 5 years may give something like 10x vote and 10x rewards but you can not withdraw your token until the 5 years are over. This is a mechanic that lends itself well to the purpose of increasing the floor over time & by locking the token the circulating supply will decrease - leading to more stability in price & rewards.
3. **Remove SBR from circulation** & reformat incentives around voting on gauges controlled by the SBR in our protocol earned from the deposited collateral and the Saber Grant
> Long term vision: In a not-so-distant future SBR rewards will dry up, and meanwhile it will probably linearly keep decreasing in price according to the increased supply and stagnant (at best) demand. If nothing is done people will just move their liquidity elsewhere, and SBR will not be able to guarantee liquidity for tokens on their platform. Mercenary farming is a short term solution, but in order to create something more sustainable we can transition the current system to one **that transfers the value in the system differently altogether.** When SBR is out of the picture, people will still have an incentive to lock their liquidity in Saber because of the fact that theres **other** things they can do with the LPs from Saber beyond farming SBR. They could mint our stablecoin, then either use it to supply liquidity on base pair pools (giving Saber $2 of TVL for every $1 in our protocol) or to purchase our governance token to control the rewards across pools (in the form of call options on the floor price of the governance token they just bought and locked for 7 days - 5 years). **Depending on how long they lock they will be providing secure liquidity from the collateral for longer in return for a greater share of the portion of rewards given to lockers, and a greater control over the portion of rewards given to liquidity pools via gauges.**

> **Users used to have only 1 way to use their CASH:** farm it against other stables or COW. **Now they will have another far more attractive option:** Use some of the CASH (or unnamed Stablecoin X in our case) to purchase the governance token that controls the treasury, which consists of growing amounts of SBR, SUNNY, and 2SBR (stablecoin LP rewards). All of these yields are providing value to the Treasury Token (which is the governance token) that votes on the permlocked SUNNY/SBR and receives liquidity constantly from the 2SBR fees paid out from Saber. On top of this it is fully backed by Stablecoin X and can only be accessed by purchasing with Stablecoin X.
 
> If there is 10% of the veSBR locked in our treasury, our governance token as a whole has the vote value of the 10% of 7m SBR emitted each week since it will control where 7m SBR are emitted. If only 1 Treasury Token is sold, it will control 100% of the 700k SBR and decide which pool it will be rewarding. In the future, the rewards will eventually flow from the SBR towards call options on the Treasury Token, which are funded by Saber fees & issue rewards across Saber stable pools. These call options will mostly go towards the remaining CASH victims, until they are made whole, with the rest being sent to veTreasury Token lockers. Once victims are paid that portion would go towards Saber pools.
4. Utilize **low risk** investment/ store of value primitives that can still facilitate risk speculation by using a floor value-enabled protocol owned market.
> At the same time, the collateral will generate % of its size in LP fees which will be paid out by adding it to the price supporting liquidity of the Treasury Token - supplementing the liquidity gained by selling tokens.
5. Lay the groundwork to going **cross-chain**. Since minting OURO and BOROS requires LPs from Saber, and BOROS must be locked into veBOROS before it receives rewards - by widening our reach and offering OURO/BOROS on different chains we can enable users to provide the Stablecoins from their chain, which could then be converted to NIRV and its pairs in order to deposit into OURO/BOROS. This would increase Saber volume since they are the gateway for most Wormhole Stable assets to be converted into something useful on Solana.
> In order to bootstrap the voting, **the Saber Team has agreed to Grant the revival project their team’s [Liquidity Reserves](https://blog.saber.so/introducing-sbr-the-saber-protocol-token-5f1f1103d57d) - which is approx. $13.6m worth of SBR at a SBR price of ~$0.016**. This will be distributed over the course of 13 months (until jul 2023) with 3 months value released at launch, then the rest linearly after 3 months. **This will bootstrap the gauges with [permalocked](https://docs.tribeca.so/features/permalock%20copy) governance power, and the only way to capture that governance power is to purchase our Treasury Token** which in turn (1) creates more governance power (SBR/SUNNY) for the system to generate from deposited collateral used to purchase govToken, (2) generates LP fees from collateral (in the form of stablecoin LPs) which are converted to price supporting liquidity, (3) captures the excess of the market price and floor price to increase price supporting liquidity even more - and eventually trigger the floor to increase.

**Thank you for taking the time to read this, and having patience with us as we put it all together. We await your feedback - particularly on [these three points](https://cashio2.notion.site/Open-Questions-For-Feedback-95c63948f6364b32a7a19db596258375).**

**Comment by tmnxeq** [comment]
As stated in Discord a clear no. 

$SBR Liquidity Reserves should not re-appropriated for ecosystem grants or to compensate losses caused by individual team member. (FWIW, despite multiple attempts, I did not get a response from the team on why they would tap the Liquidity Reserves in the first place.)
